### ISSUE TITLE
Cannot get virtual layer select dialog from provider

### ISSUE THREAD

**Issue by chleidersdorff** [qgis/QGIS#46498]
### What is the bug or the crash?

When trying to add a virtual layer, the dialogue does not open and I get the following message:
"Cannot get virtual layer select dialog from provider"

![image](https://user-images.githubusercontent.com/52182284/146131116-7eb5841b-a65c-4d5a-8625-65d8310f0d98.png)

Feature works as expected on LTR but RC has this Bug/Error.

### Steps to reproduce the issue

1. Open QGIS RC v3.22.1
2. Click "New Virtual Layer"
3. ERROR dialogue pop's up

### Versions

Here is my system information:
```
QGIS version
3.22.1-Białowieża
QGIS code revision
663dcf8fb9
Qt version
5.15.2
Python version
3.9.5
GDAL/OGR version
3.4.0
PROJ version
8.2.0
EPSG Registry database version
v10.038 (2021-10-21)
GEOS version
3.10.0-CAPI-1.16.0
SQLite version
3.35.2
PDAL version
2.3.0
PostgreSQL client version
13.0
SpatiaLite version
5.0.1
QWT version
6.1.3
QScintilla2 version
2.11.5
OS version
Windows 10 Version 1909




Active Python plugins
BulkVectorExport
1.1
changeDataSource
3.1
DataPlotly
3.8.1
geometry_paster
0.1.1
latlontools
3.5.9
LayerBoard
1.0.1
layout_loader
1.1
LoadQSS
1.4.2
loadthemall
3.3.0
MapsPrinter
0.9
mapswipetool_plugin
1.2
mmqgis
2021.9.10
poiexport
3.0.3
qfieldsync
v3.4.4
Qgis2threejs
2.6
QGIS3-getWKT
1.4
qProf
0.4.4
quicksaveqml
0.1.5
quick_map_services
0.19.27
SelectWithin
0.4
slyr_community
3.2.1
Smart_Map
1.1
spline
0.2
SuperLabeling
ver.1.4
db_manager
0.1.20
grassprovider
2.12.99
processing
2.12.99
sagaprovider
2.12.99
```

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

Only noticed it since the latest release.

**Comment by chleidersdorff** [comment]
Not sure if the solution to this problem has been implemented, but it sound similar to this: https://github.com/qgis/QGIS/issues/46116

However this issue is present on the RC version (my version) as well.

**Comment by gioman** [comment]
> Not sure if the solution to this problem has been implemented, but it sound similar to this: #46116

@chleidersdorff it is the same and it has already been fixed.

**Comment by mkoenigb** [comment]
For those who can not update their QGIS version easily, you can still use PyQGIS:

```
from qgis.core import QgsVectorLayer, QgsProject

# Create a layer
initial_query = "Select * from mylayer"
vlayer = QgsVectorLayer("?query={}".format(initial_query), "vlayer", "virtual")
QgsProject.instance().addMapLayer(vlayer)

# Update a layer
vlayer_to_update = QgsProject.instance().mapLayersByName("vlayer")[0]
updated_query = "Select id, buffer(geometry,20) as geometry from mylayer where name = 'string'"
vlayer_to_update.setDataSource("?query={}".format(updated_query), "vlayer", "virtual")
```
